include/heuristic/HeuristicMapper.hpp

```diff
@@ -60,17 +63,24 @@ class HeuristicMapper : public Mapper {
     Node() = default;
     Node(const std::array<std::int16_t, MAX_DEVICE_QUBITS>& q,
          const std::array<std::int16_t, MAX_DEVICE_QUBITS>& loc,
-         const std::vector<std::vector<Exchange>>&          sw = {}) {
+         const std::vector<std::vector<Exchange>>&          sw            = {},
+         const double                                       initCostFixed = 0) {
       std::copy(q.begin(), q.end(), qubits.begin());
       std::copy(loc.begin(), loc.end(), locations.begin());
       std::copy(sw.begin(), sw.end(), std::back_inserter(swaps));
+      costFixed = initCostFixed;
     }
 
     /**
      * @brief applies an in-place swap of 2 qubits in `qubits` and `locations`
      * of the node
      */
-    void applySWAP(const Edge& swap, Architecture& arch) {
+    void applySWAP(const Edge& swap, Architecture& arch,
+                   const std::vector<std::uint16_t> singleQubitGateMultiplicity,
+                   bool                             considerFidelity) {
+      const auto& singleQubitFidelities = arch.getSingleQubitFidelities();
+      const auto& twoQubitFidelities    = arch.getFidelityTable();
```

Comment:
I guess these statements could be moved into the if statement down below.

Reply:
Right, thank you. Will also be fixed in the next commit.